tests/utils.go

```diff
@@ -1130,7 +1130,7 @@ func NewConsoleExpecter(virtCli kubecli.KubevirtClient, vmi *v1.VirtualMachineIn
 	expecterReader, expecterWriter := io.Pipe()
 	resCh := make(chan error)
 	go func() {
-		con, err := virtCli.VirtualMachineInstance(vmi.ObjectMeta.Namespace).SerialConsole(vmi.ObjectMeta.Name)
+		con, err := virtCli.VirtualMachineInstance(vmi.Namespace).SerialConsole(vmi.Name, int(timeout))
```

Comment:
I think now that you reuse the same timeout both to wait for serial console to start AND later for the expecter, you should track how much time it took to connect to the serial console, and reduce the timeout that is used for expecter by that much. Otherwise this function may take significantly longer than expected by the caller, in case when it took significant time for serial console to connect.

Reply:
if it took a lot of time to connect the test will exit because of the test time out no?

I don't think we need to make this complex because the test have timeout for all the test scenario (create vmi, wait for console, wait for the expecter).

Reply:
Yes but your timeout semantics is now broken (in some cases): the function receives a timeout argument but then doesn't satisfy it (by taking up to twice the expected time). While indeed it probably works (=passes tests) in most cases, it's not ideal.